src/io/openshift/plugins/analytics.groovy

```diff
@@ -5,6 +5,37 @@ import io.openshift.Events
 def register() {
     Events.on("build.pass") {
         e, a -> echo "invoking bayesian analytics $e $a"
+        def image = config.runtime() ?: 'node'
+        spawn (image: image, version: config.version()) {
+          echo "Pod spawned."
+          def url = Utils.shWithOutput(this, "git config remote.origin.url")
+          echo "URL:$url"
+          image = config.runtime() ?: 'maven'
+          echo "Image:$image"
+          
+          if(image == "node") {
+              sh '''
+                  echo "running additional npm commands"
+                  npm install --prod
+                  npm list --prod --json > direct-dependencies.txt
+              '''
+          }
+          else {
+            echo "Image type not handled: $image"
```

Comment:
why do you have to spawn an image to do this check? why not exit early if the runtime is **not** node?

Reply:
I agree shouldn't we just check image used not spin it up?

Reply:
@sthaha @arilivigni This is a generic solution. Today we are implementing only for node and hence its checking only for node, when we start supporting other eco systems, their respective commands will be executed in their respective sections. Tell me a case where it will spawn a pod but will not do anything.. i cannot think of any as of now